src/rpc/client/elementHandle.ts

```diff
@@ -240,7 +243,29 @@ export function convertSelectOptionValues(values: string | ElementHandle | Selec
   return { options: values as SelectOption[] };
 }
 
-export async function convertInputFiles(files: string | FilePayload | string[] | FilePayload[]): Promise<{ name: string, mimeType: string, buffer: string }[]> {
-  const filePayloads = await normalizeFilePayloads(files);
-  return filePayloads.map(f => ({ name: f.name, mimeType: f.mimeType, buffer: f.buffer.toString('base64') }));
+type SetInputFilesFiles = ElementHandleSetInputFilesParams['files'];
+export async function convertInputFiles(files: string | FilePayload | string[] | FilePayload[]): Promise<SetInputFilesFiles> {
+  let ff: string[] | FilePayload[];
+  if (!Array.isArray(files))
+    ff = [ files ] as string[] | FilePayload[];
+  else
+    ff = files;
+  const filePayloads: SetInputFilesFiles = [];
```

Comment:
nit: file reading can be parallelized here: 

```js
const filePayloads = await Promise.all(ff.map(async ff => {
  if (typeof item === 'string') {
    return {
      name: path.basename(item),
      mimeType: mime.getType(item) || 'application/octet-stream',
      buffer: (await util.promisify(fs.readFile)(item)).toString('base64')
    };
  }
  return {
    name: item.name,
    mimeType: item.mimeType,    
    buffer: item.buffer.toString('base64'),
  };
}));

Reply:
Done.